lib/spack/spack/bootstrap.py

```diff
@@ -216,6 +216,10 @@ def try_import(self, module, abstract_spec_str):
             # specs that wwe know by dag hash.
             spack.binary_distribution.binary_index.regenerate_spec_cache()
             index = spack.binary_distribution.update_cache_and_get_specs()
+
+            if not index:
+                raise RuntimeError("Could not populate the binary index")
```

Comment:
"Partially incorporate the zen of Python" :point_up: :laughing:

Reply:
:laughing:

Reply:
Btw, I didn't know how to make the underlying error here surface... this error message is better than StopIteration for sure, but it'd be nicer to have ```Download failed: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)>``` which occurs in the `-d` output.